src/NuGet.Core/NuGet.DependencyResolver.Core/GraphModel/GraphOperations.cs

```diff
@@ -38,6 +37,32 @@ public static AnalyzeResult<RemoteResolveResult> Analyze(this GraphNode<RemoteRe
             List<DowngradeResult<RemoteResolveResult>> downgrades,
             List<GraphNode<RemoteResolveResult>> cycles)
         {
+            var workingDowngrades = RentDowngradesDictionary();
+
+            root.ForEach((node, context) => WalkTreeCheckCycleAndNearestWins(context, node), CreateState(cycles, workingDowngrades));
+
+#if NET45
```

Comment:
This can be `IS_DESKTOP`, it would future proof it incase the project changes to a higher .NET framework.

The build sets this: https://github.com/NuGet/NuGet.Client/blob/dev/build/common.targets#L5